libevm/LegacyVMCalls.cpp

```diff
@@ -219,6 +219,9 @@ bool LegacyVM::caseCallSetup(CallParameters *callParams, bytesRef& o_output)
     if (haveValueArg && m_SP[2] > 0)
         m_runGas += toInt63(m_schedule->callValueTransferGas);
 
+    if (m_ext->myAddress == destinationAddr)
+        m_runGas -= (m_schedule->callGas - m_schedule->callSelfGas);
```

Comment:
Let's better do this without subtraction, putting the condition to the initial assignment at line 207.
It's kind of simpler, no calculations, and wouldn't need to assume `callGas >= callSelfGas`